### DISCUSSION TITLE
Does Wineskin enable ESYNC by default or does it require the WINESYNC setting?  (and question about retina resolutions)

### DISCUSSION THREAD

**Discussion by sofakng** [The-Wineskin-Project/WineskinServer#376] [General]
I'm still learning about Wine but it looks like an enhancement called ESYNC is available to increase performance (but could also cause problems).

Is this enabled by default in Wineskin (v2.9.1.8) or does it require the Unix Command "export WINESYNC=1" ?

Also, is it possible to expose native/retina resolutions to the applications running inside Wine?

It looks like I need to add a registry key (RetinaMode=Y) to HKEY_CURRENT_USER\Software\Wine\Mac Driver, is that correct?

**Comment by Gcenx** [comment]
> I'm still learning about Wine but it looks like an enhancement called ESYNC is available to increase performance (but could also cause problems).

This option is more exclusive to Linux but also works on macOS via winecx.

> Is this enabled by default in Wineskin (v2.9.1.8)

It’s not enabled by default as that could break some applications/games.

> does it require the Unix Command "export WINESYNC=1" ?

Yes that’s the current requirement to enable the option.

> Also, is it possible to expose native/retina resolutions to the applications running inside Wine?
> 
> It looks like I need to add a registry key (RetinaMode=Y) to HKEY_CURRENT_USER\Software\Wine\Mac Driver, is that correct?

Before the removal of “Screen Setting” I did provide a checkbox for enabling Retina however many users just enabled it and then come complaining about the performance loss and rendering issues when it was enabled.

As wine and many windows applications don’t support hiDPI correctly restoring this option will just cause more headaches.

**Reply by sofakng** [reply]
Thanks for the information as always.

Can you explain a bit about how Wine interprets or scales the resolutions without the Retina option?  If I understand correctly, without the Retina option the games will only support a maximum render resolution of half (or lower) native screen resolution?

**Reply by Gcenx** [reply]
By default wine uses macOS native scaling, when you enable RetinaMode wine doesn’t use macOS native scaling.

Try and enabling Retina from the winetricks menu then run some windows application most will be tiny, requiring you to manually tweak the font DPI within winecfg this can cause distortions to an applications GUI.
